### ISSUE TITLE
Impossible to delete non-empty folders recursively in version 6.0.0

### ISSUE THREAD

**Issue by Dmitry1987** [jupyter/notebook#5457]
We did some research and saw that this PR was merged few years ago that was supposed to fix it: https://github.com/jupyter/notebook/pull/3108

But current version 6.0.0 does not delete non-empty folders
![image](https://user-images.githubusercontent.com/9369080/82171031-00cc2680-987b-11ea-8bfd-ff70d8a76110.png)

We run it as container spawned by jupyterhub, and here are all versions it uses:
```
ds-team@c44f1da5b31a:~$ jupyter --version
jupyter core     : 4.6.3
jupyter-notebook : 6.0.0
qtconsole        : not installed
ipython          : 7.13.0
ipykernel        : 5.2.1
jupyter client   : 6.1.3
jupyter lab      : 1.2.1
nbconvert        : 5.6.1
ipywidgets       : not installed
nbformat         : 5.0.5
traitlets        : 4.3.3
```

What am I missing, the relevant code did not change since the PR mentioned above, it is still impossible to delete.
Looks like one of the following PRs broke this functionality, might be relevant to this https://github.com/jupyter/notebook/blame/e9ce1b7461713ffccb3ca6df5ccfed3ce59b3b4b/notebook/services/contents/filemanager.py#L544

Though I am not familiar with the code base yet, just at a glance, the condition below might behave not as expected, for some reason it checks for empty/non-empty, while the code from "allow deleting non-empty folders" PR is still there in the top part of the function. 
Here is the possible issue section 
![image](https://user-images.githubusercontent.com/9369080/82171804-3a059600-987d-11ea-98a2-085dc37c8466.png)

it never gets to `rm(os_path)`

If this is by design in new versions (4.4.0 that our team use on old clusters, can delete non-empty) would it be possible to put this under a feature-flag to allow optional behavior for those who need it?

**Comment by Dmitry1987** [comment]
Closing issue, because resolved after discussion here https://github.com/jupyter/notebook/issues/4916

TLDR for google search landers: one of the options is to patch that file if you don't care about the "deleting to trash" feature, just remove it from there and make the code go directly to deletion steps, here's the modified function from notebook==6.0.3:

```
    def delete_file(self, path):
        """Delete file at path."""
        path = path.strip('/')
        os_path = self._get_os_path(path)
        rm = os.unlink
        if not os.path.exists(os_path):
            raise web.HTTPError(404, u'File or directory does not exist: %s' % os_path)

        if os.path.isdir(os_path):
            self.log.debug("Removing directory %s", os_path)
            with self.perm_to_403():
                shutil.rmtree(os_path)
        else:
            self.log.debug("Unlinking file %s", os_path)
            with self.perm_to_403():
                rm(os_path)
```
